controllers/pkg/reconcilers/repository/reconciler_test.go

```diff
@@ -16,187 +16,260 @@ package repository
 
 import (
 	"context"
-	"errors"
+	"fmt"
 	"testing"
 
+	"github.com/go-logr/logr"
+	"github.com/nephio-project/nephio/controllers/pkg/giteaclient"
+	"github.com/nephio-project/nephio/controllers/pkg/resource"
+	"github.com/stretchr/testify/mock"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
 	"code.gitea.io/sdk/gitea"
 	infrav1alpha1 "github.com/nephio-project/api/infra/v1alpha1"
-	"github.com/stretchr/testify/require"
 )
 
-type GiteaClientMock struct {
-}
-
-type NephioGiteaClientMock struct {
-	myUserInfoError error
-	deleteRepoError error
-	getRepoError    error
-	createRepoError error
-	editRepoError   error
-}
-
-func (gc NephioGiteaClientMock) IsInitialized() bool {
-	return true
-}
-
-func (gc NephioGiteaClientMock) Get() *gitea.Client {
-	return nil
-}
-
-func (gc NephioGiteaClientMock) Start(ctx context.Context) {
-}
-
-func (gc NephioGiteaClientMock) GetMyUserInfo() (*gitea.User, *gitea.Response, error) {
-	return &gitea.User{}, &gitea.Response{}, gc.myUserInfoError
+type mockHelper struct {
```

Comment:
Suggestion: it may be a good idea to create a separate directory called testing and defined the mockHelper struct there and then could be imported by other packages that will use mockery for unit testing

Reply:
Done.